frontend/express/public/javascripts/countly/countly.views.js

```diff
@@ -638,410 +638,186 @@ window.UserView = countlyView.extend({
+    data: function() {
+        return {
+            barChartItemsColors: {
+                all: "#017AFF",
+                sevenDays: "#F96300",
+                thirtyDays: "#39C0C8"
+            },
+            barChartItemsLegends: {
+                all: CV.i18n('user-activity.barchart-all-users'),
+                sevenDays: CV.i18n('user-activity.barchart-seven-days'),
+                thirtyDays: CV.i18n('user-activity.barchart-thirty-days')
+            },
         };
-
-        // Datatable data
-        var chartData = [];
-        for (var iTick = 1; iTick < ticks.length - 1; iTick++) {
-            var all = dp[0].data[iTick][1] ? dp[0].data[iTick][1] : 0;
-            var allPercentage = countlyCommon.formatNumber((100 * all) / totals[0], 2);
-            allPercentage = isNaN(allPercentage) ? 0 : allPercentage;
-
-            var tDays = (dp[1].data[iTick] && dp[1].data[iTick][1]) ? dp[1].data[iTick][1] : 0;
-            var tDaysPercentage = countlyCommon.formatNumber((100 * tDays) / totals[1], 2);
-            tDaysPercentage = isNaN(tDaysPercentage) ? 0 : tDaysPercentage;
-
-            var sDays = (dp[2].data[iTick] && dp[2].data[iTick][1]) ? dp[2].data[iTick][1] : 0;
-            var sDaysPercentage = countlyCommon.formatNumber((100 * sDays) / totals[2], 2);
-            sDaysPercentage = isNaN(sDaysPercentage) ? 0 : sDaysPercentage;
-
-            var allMultiplier = 0.8;
-            var tDaysMultiplier = 0.8;
-            var sDaysMultiplier = 0.8;
-
-            if (allPercentage > 80) {
-                allMultiplier = 0.65;
-            }
-
-            if (tDaysPercentage > 80) {
-                tDaysMultiplier = 0.65;
-            }
-
-            if (sDaysPercentage > 80) {
-                sDaysMultiplier = 0.65;
-            }
-
-            chartData.push({
-                l: ticks[iTick][1],
-                a_count: all,
-                td_count: tDays,
-                sd_count: sDays,
-                a: "<div style='float:left;min-width: 40px'>" + countlyCommon.formatNumber(all) + "</div><div class='percent-bar' style='width:" + (allPercentage * allMultiplier) + "%'></div>" + allPercentage + "%",
-                td: "<div style='float:left;min-width: 40px'>" + countlyCommon.formatNumber(tDays) + "</div><div class='percent-bar' style='width:" + (tDaysPercentage * tDaysMultiplier) + "%'></div>" + tDaysPercentage + "%",
-                sd: "<div style='float:left;min-width: 40px'>" + countlyCommon.formatNumber(sDays) + "</div><div class='percent-bar' style='width:" + (sDaysPercentage * sDaysMultiplier) + "%'></div>" + sDaysPercentage + "%"
+    },
+    computed: {
+        userActivity: function() {
+            return this.$store.state.countlyUserActivity.userActivity;
+        },
+        seriesTotal: function() {
+            return this.$store.state.countlyUserActivity.seriesTotal;
+        },
+
+        userActivityOptions: function() {
+            return {
+                legend: {
```

Comment:
charts automatically has a default legend setting.
unless you want to change that, no need to pass legend options explicitly.